javascript/sdk/corvy.js

```diff
@@ -1,251 +1,354 @@
 
-  async login() {
-    try {
-      const res = await this.client.post("/auth");
-      this.user = res.data.bot;
+        this.socket.connect();
 
-      await this.flocks.fetch();
+        this.socket.onOpen(() => {
+            this._log('✅ Socket connection established');
+            this.emit('ready', this.bot);
+        });
+        this.socket.onError((error) => {
+            this._log(`❌ Socket error: ${error.message}`);
+            this.emit('error', new Error(`Socket error: ${error.message}`));
+        });
+        this.socket.onClose((event) => this._log(`Socket connection closed: ${event.code} - ${event.reason}`));
 
-      this.emit("ready", this);
+        this._joinBotChannel();
+    }
 
-      this.startTime = Date.now();
+    _joinBotChannel() {
+        this.channel = this.socket.channel("bot:any", { token: this.token });
 
-      const baseline = await this.client.get("/messages", {
-        params: { cursor: 0 }
-      });
+        this.channel.on("message", (payload) => {
+            if (payload.event === 'new_message') {
+                this._handleNewMessage(payload.message);
+            }
+        });
 
-      this.currentCursor = baseline.data.cursor || 0;
+        this.channel.join()
+            .receive("ok", (response) => {
+                this._log('✅ Successfully joined bot channel');
+            })
+            .receive("error", (response) => {
+                this._log(`❌ Failed to join bot channel: ${JSON.stringify(response)}`);
+                this.emit('error', new Error(`Failed to join bot channel: ${JSON.stringify(response)}`));
+            });
+    }
 
-      this._processMessages();
+    _setupProcessHandlers() {
+        process.on('SIGINT', () => {
+            this._log('\nShutting down...');
+            if (this.socket) {
+                this.socket.disconnect();
+            }
+            process.exit(0);
+        });
+    }
 
-      process.on("SIGINT", () => {
-        console.log("Client shutting down...");
-        process.exit(0);
-      });
-    } catch (err) {
-      this.emit("error", new Error("Error logging in: " + err.message));
-      this._log(`Error logging in: ${err.message}`);
+    checkForStaff(userid, staffs) {
+        return staffs?.includes(userid)
     }
-  }
 
-  // -------------------------------
-  // Message Loop
-  // -------------------------------
+    registerCommand(data, handler) {
+        const names = data.names ?? data
+        const desc = data.desc
+        const staffUserIds = data.staff
+        if (!handler || typeof handler !== 'function') {
+            throw new Error('Command handler must be a function');
+        }
 
-  async _processMessages() {
-    this._log("Starting message loop...");
+        const nameList = Array.isArray(names) ? names : [names];
+
+        for (const name of nameList) {
+            if (typeof name !== 'string' || name.trim() === '') {
+                throw new Error('Command name must be a non-empty string');
```

Comment:
not sure if you can do this in js, but if you could it might be a good idea to fall back to the function name

Reply:
Not possible. `handler` will most likely be an anonymous function anyway which means no function name. Even if so I dont think adding safeguards for that would be worth it at all.

Reply:
fair enough